airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -363,33 +390,61 @@ def stream_slices(
         ticket_stream_value = stream_state.get(Tickets.cursor_field)
         if not ticket_stream_value:
             # for backward compatibility because not all relevant states can have some last ticket state
-            ticket_stream_value = Tickets.str2unixtime(stream_state.get(self.cursor_field))
-
-        tickets = Tickets(start_date=self._start_date, subdomain=self._subdomain, authenticator=self.authenticator).read_records(
-            sync_mode=sync_mode, cursor_field=cursor_field, stream_state={Tickets.cursor_field: ticket_stream_value}
-        )
-
-        # selects all tickets what have at least one comment
-        ticket_ids = [
-            {
-                "id": ticket["id"],
-                Tickets.cursor_field: ticket[Tickets.cursor_field],
-            }
-            for ticket in tickets
-            if ticket["comment_count"]
-        ]
-        self.logger.info(f"Found updated {len(ticket_ids)} ticket(s) with comments")
-        # sort slices by generated_timestamp
-        ticket_ids.sort(key=lambda ticket: ticket[Tickets.cursor_field])
-        return ticket_ids
+            ticket_stream_value = self.str2unixtime(stream_state.get(self.cursor_field))
+
+        tickets_stream = Tickets(start_date=self._start_date, subdomain=self._subdomain, authenticator=self.authenticator)
+        ticket_pages = defaultdict(list)
+        last_end_time = stream_state.get(LAST_END_TIME_KEY, 0)
+        ticket_count = 0
+        for ticket in tickets_stream.read_records(
+            sync_mode=sync_mode,
+            cursor_field=cursor_field,
+            stream_state={Tickets.cursor_field: ticket_stream_value, LAST_END_TIME_KEY: last_end_time},
+        ):
+            if not ticket["comment_count"]:
+                # skip tickets without comments
+                continue
+            ticket_count += 1
+            ticket_pages[tickets_stream.last_end_time].append(
+                {
+                    "id": ticket["id"],
+                    Tickets.cursor_field: ticket[Tickets.cursor_field],
+                }
+            )
+
+        if ticket_pages:
+            last_times = sorted(ticket_pages.keys())
```

Comment:
based on doing `ticket_pages[tickets_stream.last_end_time].append(...` above, won't ticket_pages just have one unique key which is `Tickets(start_date=self._start_date, subdomain=self._subdomain, authenticator=self.authenticator).last_end_time`?

This code seems like it expects multiple...

Reply:
(relevant down to line 430)